### PR TITLE
Add phpstan level 0

### PR DESCRIPTION
**Pull request by franmomu**
This PR tries to add PHPStan to the package, in order to reach level 0, the changes I've made are about:
- Add missing return statements
- Remove unused arguments and parameters

I've ignored errors coming from not existing `phpiredis_*` functions,

### CODE REVIEW

File: phpstan-baseline.neon [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+		-
+			message: "#^Static method Predis\\\\Command\\\\RawCommand\\:\\:create\\(\\) invoked with 0 parameters, at least 1 required\\.$#"
+			count: 1
+			path: tests/Predis/Command/RawCommandTest.php
+
+		-
+			message: "#^Call to an undefined method Predis\\\\Configuration\\\\Option\\\\AggregateTest\\:\\:getMockConnectionClass\\(\\)\\.$#"
+			count: 2
+			path: tests/Predis/Configuration/Option/AggregateTest.php
```
Comment (franmomu) [inline phpstan-baseline.neon:21]: These methods are weird: https://github.com/predis/predis/blob/113145f3065b8f34da2e09359f403e4db213b136/tests/Predis/Configuration/Option/AggregateTest.php#L258-L309

File: phpstan-baseline.neon [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+			count: 3
+			path: src/Client.php
+
+		-
+			message: "#^Unsafe usage of new static\\(\\)\\.$#"
+			count: 1
+			path: src/Connection/Parameters.php
+
+		-
+			message: "#^Static method Predis\\\\Command\\\\RawCommand\\:\\:create\\(\\) invoked with 0 parameters, at least 1 required\\.$#"
```
Comment (franmomu) [inline phpstan-baseline.neon:14]: This is on purpose to test the exception:

https://github.com/predis/predis/blob/113145f3065b8f34da2e09359f403e4db213b136/tests/Predis/Command/RawCommandTest.php#L66-L71

File: phpstan.neon [hunk 3]
```diff
@@ -0,0 +1,10 @@
+includes:
```
Comment (tillkruss) [inline phpstan.neon:1]: Can we merge the two config files and call it `phpstan.dist.neon`?

File: composer.json [hunk 4]
```diff
@@ -31,7 +31,8 @@
         "php": "^7.2 || ^8.0"
     },
     "require-dev": {
-        "phpunit/phpunit": "^8.0 || ~9.4.4"
+        "phpunit/phpunit": "^8.0 || ~9.4.4",
+        "phpstan/phpstan": "^1.9"
```
Comment (tillkruss) [inline composer.json:35]: Can we install it via `shivammathur/setup-php` without making it a dependency?

File: .github/workflows/static-analysis.yml [hunk 5]
```diff
@@ -0,0 +1,56 @@
+name: Static Analysis
```
Comment (tillkruss) [inline .github/workflows/static-analysis.yml:1]: Let's just add the job to the `coding-standards` workflow, that was just merged.

**Review by tillkruss — CHANGES_REQUESTED** [review]
Great job!

**Comment by tillkruss** [comment]
> I've ignored errors coming from not existing `phpiredis_*` functions

Yeah, unless we can get phpiredis migrated over, we'll likely remove it in favor of Relay anyhow.

**Review by tillkruss — APPROVED** [review]

**Pull request merged**